apps/managedidentity/managedidentity.go

```diff
@@ -323,9 +360,56 @@ func authResultFromToken(authParams authority.AuthParams, token accesstokens.Tok
 	return ar, err
 }
 
+// Contains checks if the element is present in the list.
+func contains[T comparable](list []T, element T) bool {
+	for _, v := range list {
+		if v == element {
+			return true
+		}
+	}
+	return false
+}
+
+// retry performs an HTTP request with retries based on the provided options.
+func retry(maxRetries int, c ops.HTTPClient, req *http.Request, s Source) (*http.Response, error) {
+	var resp *http.Response
+	var err error
+	for attempt := 0; attempt < maxRetries; attempt++ {
+		tryCtx, tryCancel := context.WithTimeout(req.Context(), time.Second*15)
+		defer tryCancel()
+		cloneReq := req.Clone(tryCtx)
+		resp, err = c.Do(cloneReq)
+		retrylist := retryStatusCodes
+		if s == DefaultToIMDS {
+			retrylist = append(retrylist, retryCodesForIMDS...)
+		}
```

Comment:
Note that on an IMDS platform, every retry allocates and discards enough memory to hold `retryStatusCodes` and `retryCodesForIMDS`. It isn't much memory but this makes unnecessary work for the GC. There's no need to recompute this slice because the codes don't change between iterations of this loop or even between calls to this function. Why not simply have `retryCodesForIMDS` hold all the codes for IMDS and `retryStatusCodes` hold all the codes for other sources?